k4MarlinWrapper/src/components/EDM4hep2Lcio.cpp

```diff
@@ -282,11 +282,27 @@ void EDM4hep2LcioTool::convertEventHeader(const std::string& e4h_coll_name, lcio
 void EDM4hep2LcioTool::convertAdd(const std::string& e4h_coll_name, const std::string& lcio_coll_name,
                                   lcio::LCEventImpl* lcio_event, CollectionPairMappings& collection_pairs,
                                   std::vector<EDM4hep2LCIOConv::ParticleIDConvData>& pidCollections) {
-  const auto& evtFrame = m_podioDataSvc->getEventFrame();
-  const auto& metadata = m_podioDataSvc->getMetaDataFrame();
-  const auto  collPtr  = evtFrame.get(e4h_coll_name);
-  if (!collPtr) {
-    error() << "No collection with name: " << e4h_coll_name << " available for conversion" << endmsg;
+  const auto&            metadata = m_podioDataSvc->getMetaDataFrame();
+  podio::CollectionBase* collPtr  = nullptr;
+  DataObject*            p;
+  auto                   sc = m_podioDataSvc->retrieveObject(e4h_coll_name, p);
+  if (sc.isFailure()) {
+    throw GaudiException("Collection not found", name(), StatusCode::FAILURE);
+  }
+  auto ptr = dynamic_cast<DataWrapperBase*>(p);
+  if (ptr) {
+    collPtr = ptr->collectionBase();
+  }
```

Comment:
Slightly more verbose than before, but at least we got the detail of using a Frame out of here.